earn/src/components/borrow/BorrowMetrics.tsx

```diff
@@ -119,45 +122,110 @@ function HealthMetricCard(props: { health: number }) {
 }
 
 export type BorrowMetricsProps = {
-  marginAccountPreview?: MarginAccountPreview;
+  marginAccount?: MarginAccount;
   dailyInterest0: number;
   dailyInterest1: number;
 };
 
 export function BorrowMetrics(props: BorrowMetricsProps) {
-  const { marginAccountPreview, dailyInterest0, dailyInterest1 } = props;
-  if (!marginAccountPreview) {
-    return null;
+  const { marginAccount, dailyInterest0, dailyInterest1 } = props;
+
+  const maxSafeCollateralFall = useMemo(() => {
+    if (!marginAccount) return null;
+
+    const { lowerSqrtRatio, upperSqrtRatio } = computeLiquidationThresholds(
+      marginAccount.assets,
+      marginAccount.liabilities,
+      [], // TODO: We should actually fetch Uniswap positions
```

Comment:
Is all we need to do fetch these and pass them in?

Reply:
Yes, I just wasn't sure where to put the fetching logic. Right now this is the only place we'd need it, but if we allow users to user uniswap positions as collateral, we'd want to fetch this in the outer page component so that we can do other things with it (e.g. display a table of positions)